notebook/notebookapp.py

```diff
@@ -429,20 +429,10 @@ def start(self):
         self.log.info("Wrote hashed password to %s" % self.config_file)
 
 
-def shutdown_server(server_info, timeout=5, log=None):
-    """Shutdown a notebook server in a separate process.
-
-    *server_info* should be a dictionary as produced by list_running_servers().
-
-    Will first try to request shutdown using /api/shutdown .
-    On Unix, if the server is still running after *timeout* seconds, it will
-    send SIGTERM. After another timeout, it escalates to SIGKILL.
-
-    Returns True if the server was stopped by any means, False if stopping it
-    failed (on Windows).
-    """
+def kernel_request(server_info, path='/login', method='GET', body=None, headers=None, timeout=5, log=None):
```

Comment:
Per the previous conversation, this method name does not reflect its function and `path` should not have a default since it's used in exactly two locations, neither of which use `'/login'`.
```suggestion
def server_request(server_info, path, method='GET', body=None, headers=None, timeout=5, log=None):
```